app/helpers/suffix_tree_helper.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'singleton'
+
+module SuffixTreeHelper
+  class SuffixTreeSingleton < SuffixTree
+    include Singleton
+    def initialize
+      super(AppConfig['suffix_tree']['str_path'], AppConfig['suffix_tree']['tag_path'],
+            AppConfig['suffix_tree']['child_path'], AppConfig['suffix_tree']['node_path'])
+      @functional = true
+      @reason = ''
+    end
+
+    def mark_functional
+      @functional = true
+    end
+
+    def mark_broken(reason)
+      @functional = false
+      @reason = reason
+    end
+
+    def functional?
+      @functional
+    end
+
+    def broken_reason
+      @reason
+    end
+  end
+
+  FIELD_BITMASK = {
+    title: 1,
+    body: 2,
+    why: 4,
+    username: 8
+  }.freeze
+
+  def self.available_fields
+    FIELD_BITMASK.keys
+  end
+
+  def self.calc_mask(fields)
```

Comment:
You'll prolly want to intersect `fields` with `available_fields` to avoid `nil` related errors

Reply:
I *think* I have already done the intersection before calling calc_mask...

Reply:
My feeling is that it makes more logical sense to do that here, every single caller of this method will want the intersect and so that logic should live in this method.